Oleksandr.Svidovyi/Task_5/src/main/java/com/hillel/task_5/servlet/AccountServlet.java

```diff
@@ -0,0 +1,178 @@
+            }
+
+        } catch (Exception ex) {
+            resp.getWriter().println("Something went wrong. Please, try again.");
+            resp.setStatus(400);
+        }
+    }
+
+
+    @Override
+    protected void doPut(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+        resp.setContentType("application/json");
+        Gson gson = new Gson();
+        clientRepository = ClientRepository.getInstance();
+
+        try {
+            AccountPostRequestModel reqeustModel = gson.fromJson(req.getReader(), AccountPostRequestModel.class);
+
+            if (reqeustModel.getBalance() != null && !reqeustModel.getBalance().equals("")
+                    && reqeustModel.getCurrency() != null && !reqeustModel.getCurrency().equals("")
+                    && reqeustModel.getClientId() != null && !reqeustModel.getClientId().equals("")) {
+                Integer clientId = Integer.decode(reqeustModel.getClientId());
+                Long balance = Long.decode(reqeustModel.getBalance());
+                Account account = new Account(clientId, reqeustModel.getCurrency(), balance);
+
+                if (clientRepository.getClients().get(clientId) != null
+                        && clientRepository.getClients().get(clientId).getAccount() != null) {
+                    clientRepository.getClients().get(clientId).setAccount(account);
+
+                    resp.getWriter().println("CLient's account was edited \n"
+                            + gson.toJson(clientRepository.getClients().get(clientId)));
+
+                } else {
+                    resp.getWriter().println("Account not found.");
+                    resp.setStatus(404);
+
+                }
+            } else {
+                resp.getWriter().println("Please, enter account's attributes");
+                resp.setStatus(400);
+            }
+
+        } catch (Exception ex) {
+            resp.getWriter().println("Something went wrong. Please, try again.");
+            resp.setStatus(400);
+        }
+    }
+
+
+    @Override
+    protected void doDelete(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+        resp.setContentType("application/json");
+        Gson gson = new Gson();
+        clientRepository = ClientRepository.getInstance();
+
+        try {
+            AccountGetRequestModel reqeustModel = gson.fromJson(req.getReader(), AccountGetRequestModel.class);
+
+            if (reqeustModel.getClientId() != null && !reqeustModel.getClientId().equals("")) {
+                Integer clientId = Integer.decode(reqeustModel.getClientId());
+
+                if (clientId <= clientRepository.getClients().size()) {
+
+                    if (clientRepository.getClients().get(clientId) != null
+                            && clientRepository.getClients().get(clientId).getAccount() != null) {
+                        clientRepository.getClients().get(clientId).setAccount(new Account());
```

Comment:
It's bad approach

Create method like clientRepository.addAccount(clientId, account)

Reply:
Done.